File: TestAppBarLayout/app/src/main/java/com/pix/testappbarlayout/MainActivity.java
package com.pix.testappbarlayout;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void scrollTypeClick(View v) {
        startActivity(new Intent(this,ScrollTypeActivity.class));
    }
    public void enterAlwaysClick(View v) {
        startActivity(new Intent(this,EnterAlwaysActivity.class));
    }
    public void eucClick(View v) {
        startActivity(new Intent(this,EUCTypeActivity.class));
    }
    public void eacClick(View v) {
        startActivity(new Intent(this,EACTypeActivity.class));
    }
    public void ctlClick(View v) {
        startActivity(new Intent(this,TestCollapsingToolbarLayoutActivity.class));
    }
}

File: TestShape/shape-code/src/main/java/com/pix/testshape/MainActivity.kt
package com.pix.testshape

import android.graphics.Color
import android.graphics.drawable.GradientDrawable
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.pix.testshape.ext.btn_shape_selector
import com.pix.testshape.ext.dp
import com.pix.testshape.ext.pressDrawableSelector
import com.pix.testshape.ext.shape
import kotlinx.android.synthetic.main.activity_main.*
import org.w3c.dom.Text

/**
 * 展示shape的Demo
 */
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initView()
    }

    private fun initView() {
        // 构建GradientDrawable对象并设置属性
        val drawable = GradientDrawable().apply {
            shape = GradientDrawable.RECTANGLE // 矩形
            cornerRadius = 5.dp.toFloat() // 圆角
            colors = intArrayOf(Color.parseColor("#ff00ff"),Color.parseColor("#800000ff")) //渐变色
            gradientType = GradientDrawable.LINEAR_GRADIENT // 渐变类型
            orientation = GradientDrawable.Orientation.LEFT_RIGHT // 渐变方向
            setStroke(2.dp,Color.parseColor("#ffff00")) // 描边宽度和颜色
        }
        btn_codeshape.background = drawable
        btn_dslshape.background = shape {
            shape = GradientDrawable.RECTANGLE
            cornerRadius = 10f
            colors = intArrayOf(Color.parseColor("#ff00ff"),Color.parseColor("#800000ff"))
            gradientType = GradientDrawable.LINEAR_GRADIENT
            orientation = GradientDrawable.Orientation.LEFT_RIGHT
            setStroke(2.dp,Color.parseColor("#ffff00"))
        }

        var shapeNormal = shape {
            shape = GradientDrawable.RECTANGLE
            cornerRadius = 10f
            setColor(Color.BLUE)
            setStroke(2.dp,Color.parseColor("#ffff00"))
        }

        var shapePresssed = shape {
            shape = GradientDrawable.RECTANGLE
            cornerRadius = 10f
            setColor(Color.GRAY)
            setStroke(2.dp,Color.parseColor("#ffff00"))
        }

//        btn_selector.background = btn_shape_selector(shapeNormal,shapePresssed)

        btn_selector.background = pressDrawableSelector {
            pressedDrawable = shape {
                shape = GradientDrawable.RECTANGLE
                cornerRadius = 10f
                setColor(Color.BLUE)
                setStroke(2.dp,Color.parseColor("#ffff00"))
            }
            unpressedDrawable = shape {
                shape = GradientDrawable.RECTANGLE
                cornerRadius = 10f
                setColor(Color.GRAY)
                setStroke(2.dp,Color.parseColor("#ffff00"))
            }
        }

    }
}